src/ray/stats/metric_exporter.h

```diff
@@ -100,38 +102,83 @@ class OpenCensusProtoExporter final : public opencensus::stats::StatsExporter::H
+                          size_t report_batch_size,
+                          size_t max_grpc_payload_size);
+
+  OpenCensusProtoExporter(std::shared_ptr<rpc::MetricsAgentClient> agent_client,
+                          const WorkerID &worker_id,
+                          size_t report_batch_size,
+                          size_t max_grpc_payload_size);
 
   ~OpenCensusProtoExporter() = default;
 
   static void Register(const int port,
                        instrumented_io_context &io_service,
                        const std::string address,
                        const WorkerID &worker_id,
-                       size_t report_batch_size) {
+                       size_t report_batch_size,
+                       size_t max_grpc_payload_size) {
     opencensus::stats::StatsExporter::RegisterPushHandler(
-        absl::make_unique<OpenCensusProtoExporter>(port, io_service, address, worker_id, report_batch_size));
+        absl::make_unique<OpenCensusProtoExporter>(port,
+                                                   io_service,
+                                                   address,
+                                                   worker_id,
+                                                   report_batch_size,
+                                                   max_grpc_payload_size));
   }
 
   void ExportViewData(
       const std::vector<std::pair<opencensus::stats::ViewDescriptor,
                                   opencensus::stats::ViewData>> &data) override;
+  void SendData(const rpc::ReportOCMetricsRequest &request);
+
+  /// Processes data from the provided ViewData container by
+  ///   - Adding it into corresponding proto request payload
+  ///   - Submitting request payload to agent client (when it's reached target batch size
+  ///   or
+  ///     payload size limits)
+  ///
+  /// Upon returning of this method all of the time-series from the provided ViewData
+  /// container will either be a) contained inside provide proto request payload or b)
+  /// already submitted to the client
+  ///
+  /// \param view_descriptor, descriptor of the metric
+  /// \param view_data, data container aggregating time-series for this metric (across
+  /// different set of tags)
+  /// \param request_proto, target proto payload to embed metric values into
+  /// \param cur_batch_size current size of the batch (in terms of number of time-series
+  /// already added)
+  /// \param next_payload_size_check_at next batch size when payload (binary) size have
+  /// to be checked
+  void ProcessMetricsData(const opencensus::stats::ViewDescriptor &view_descriptor,
+                          const opencensus::stats::ViewData &view_data,
+                          rpc::ReportOCMetricsRequest &request_proto,
+                          size_t &cur_batch_size,
+                          size_t &next_payload_size_check_at);
+
+ protected:
+  rpc::ReportOCMetricsRequest createRequestProtoPayload();
```

Comment:
can you add docstring for these methods?

Reply:
These are internal methods (abstracting common boiler-plate) that are not meant to be used externally and on top of it these are also quite straightforward so not sure if it's worthwhile to add doc-comments for them